### PR TITLE
Update installing from source to include 2 wasm32 targets.

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

Since (and including) Spin's [eb8239756e85d3c1bfd4d9e4b94bb4f6a0bb4fe4](https://github.com/fermyon/spin/commit/eb8239756e85d3c1bfd4d9e4b94bb4f6a0bb4fe4) commit, the `wasm32-unknown-unknown` is required in order for Spin to build from source. In addition, the Rust version required is `1.67.0` and above/newer.

Tested with both `make build` and `cargo` on main at 

```bash
$ cd ~
$ git clone https://github.com/fermyon/spin
$ cd spin
$ rustup target add wasm32-wasi
$ rustup target add wasm32-unknown-unknown
$ make build
    cargo build --release
    Compiling spin-cli v1.1.0-pre0 (/home/tpmccallum/spin)
    Finished release [optimized] target(s) in 53.91s
```

### CODE REVIEW

**Comment by tpmccallum** [comment]
In addition to checking Rust versions for installing Spin, a check was also done to test the Minimum Supported Rust Version (MSRV) in relation to just building and running a Spin application itself. Seems that Rust 1.64.0 is still a valid MSRV in relation to `spin build` and `spin up` (at least for this http-rust example anyway).

```bash
# Downgraded Rust version to 1.64.0
$ rustup default 1.64.0
1.64.0-x86_64-unknown-linux-gnu installed - rustc 1.64.0 (a55dd71d5 2022-09-19)

# Added wasm32-wasi target
$rustup target add wasm32-wasi

# Upgrade template command
$ spin templates upgrade

# Created new http rust application
$ spin new http-rust
Enter a name for your new application: test_msrv
Project description: Test minimum supported rust version
HTTP base: /
HTTP path: /...
$ cd test_msrv

# Updated Cargo.toml to point to Spin main 
# vi Cargo.toml
# The Spin SDK.
spin-sdk = { git = "https://github.com/fermyon/spin", branch = "main" }

# Build the Spin app
$ spin build
Successfully ran the build command for the Spin components.

# Run Spin app
$ spin up
Serving http://127.0.0.1:3000/ 

# Response - ok
"Hello, Fermyon"
```

File: content/spin/install.md [hunk 1]
```diff
@@ -138,12 +138,13 @@ If you have [`cargo`](https://doc.rust-lang.org/cargo/getting-started/installati
...
 <pre class="bash spin-install" id="spin-install-source-cargo"><code>$ git clone https://github.com/fermyon/spin -b v1.0.0
 $ cd spin
+$ rustup target add wasm32-unknown-unknown
 $ rustup target add wasm32-wasi
 $ cargo install --locked --path .
 $ spin --help
 </code></pre>
 
-> Please note: Spin v1.0.0 requires `wasmtime v5.0.0` which requires rustc 1.66.0 or newer. You can update Rust using the following command:
+> Please note: Spin v1.0.0 requires `wasmtime v5.0.0` which requires rustc 1.67.0 or newer. You can update your version of Rust using the following command:
```
Comment (itowlson) [inline content/spin/install.md:147]: Is it true that the 1.67 requirement is down to Wasmtime 5 (i.e. the old Rust version was wrong), or is there something else that is now forcing us to Rust 1.67 (i.e. the reason for the new Rust version is different).

It might be better to omit the reason and say only "Spin 1.1 requires Rust 1.67 or above."  Not sure users care which dependency is causing the minimum version requirement.

**Review by itowlson — CHANGES_REQUESTED** [review]
